node/pkg/evm/connectors/logpoller.go

```diff
@@ -0,0 +1,158 @@
+	connector := &LogPollConnector{Connector: baseConnector, client: client}
+	// The supervisor will keep the poller running
+	err := supervisor.Run(ctx, "logPoller", connector.run)
+	if err != nil {
+		return nil, err
+	}
+	return connector, nil
+}
+
+func (l *LogPollConnector) run(ctx context.Context) error {
+	logger := supervisor.Logger(ctx).With(zap.String("eth_network", l.Connector.NetworkName()))
+
+	blockChan := make(chan *NewBlock)
+	sub, err := l.SubscribeForBlocks(ctx, blockChan)
+	if err != nil {
+		return err
+	}
+	defer sub.Unsubscribe()
+
+	supervisor.Signal(ctx, supervisor.SignalHealthy)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case err := <-sub.Err():
+			return err
+		case block := <-blockChan:
+			err := l.processBlock(ctx, logger, block)
+			if err != nil {
+				l.errFeed.Send(err)
+			}
+		}
+	}
+}
+
+func (l *LogPollConnector) WatchLogMessagePublished(ctx context.Context, sink chan<- *ethAbi.AbiLogMessagePublished) (ethEvent.Subscription, error) {
+	sub := &PollSubscription{
+		err: make(chan error, 1),
+	}
+
+	innerSink := make(chan *ethAbi.AbiLogMessagePublished, 10)
+	innerSub := l.messageFeed.Subscribe(innerSink)
+
+	// The feed library does not support error forwarding, so we're emulating that using a custom subscription and
+	// an error feed.
+	innerErrSink := make(chan error, 10)
+	innerErrSub := l.errFeed.Subscribe(innerErrSink)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				innerSub.Unsubscribe()
+				innerErrSub.Unsubscribe()
+				return
+			case <-sub.quit:
+				innerSub.Unsubscribe()
+				innerErrSub.Unsubscribe()
+				sub.unsubDone <- struct{}{}
+				return
+			case v := <-innerSink:
+				sink <- v
+			case v := <-innerErrSink:
+				sub.err <- v
+			}
+		}
+	}()
+	return sub, nil
+}
+
+var (
+	getLogsBigOne       = big.NewInt(1)
+	logsLogMessageTopic = ethCommon.HexToHash("0x6eb224fb001ed210e379b335e35efe88672a8ce935d981a6896b27ffdf52a3b2")
+)
+
+func (l *LogPollConnector) processBlock(ctx context.Context, logger *zap.Logger, block *NewBlock) error {
+	if l.prevBlockNum == nil {
+		l.prevBlockNum = new(big.Int).Set(block.Number)
+	} else {
+		l.prevBlockNum.Add(l.prevBlockNum, getLogsBigOne)
+	}
+
+	filter := ethereum.FilterQuery{
+		FromBlock: l.prevBlockNum,
+		ToBlock:   block.Number,
+		Addresses: []ethCommon.Address{l.ContractAddress()},
+	}
+
+	*l.prevBlockNum = *block.Number
+
+	tCtx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	logs, err := l.client.FilterLogs(tCtx, filter)
+	if err != nil {
+		logger.Error("GetLogsQuery: query of eth_getLogs failed",
+			zap.Stringer("FromBlock", filter.FromBlock),
+			zap.Stringer("ToBlock", filter.ToBlock),
+			zap.Error(err),
+		)
+
+		return fmt.Errorf("GetLogsQuery: failed to query for log messages: %w", err)
+	}
+
+	if len(logs) == 0 {
+		return nil
+	}
+
+	for _, log := range logs {
+		if log.Topics[0] == logsLogMessageTopic {
```

Comment:
This is more of a personal preference but we can invert this condition to reduce nesting, which I personally find to be more readable:

```
for _, log := range logs {
	if log.Topics[0] != logsLogMessageTopic {
		continue
	}

	ev, err := l.ParseLogMessagePublished(log)
	if err != nil {
		// logging
		continue
	}

	l.messageFeed.Send(ev)
}
```

This makes the normal code path have the lowest indentation and only the exceptional paths have extra indentation. Also see https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow